lib/models/api_calls.dart

```diff
@@ -51,24 +51,27 @@ class ApiCalls {
   getDeviceList(String authToken, String nextPageToken) async {
     var response;
     authToken = 'Bearer $authToken';
-    try {
-      nextPageToken == null
-          ? response = await _client.get(DEVICE_LIST_URL,
-              headers: {HttpHeaders.authorizationHeader: authToken})
-          : response = await _client.get(
-              DEVICE_LIST_WITH_TOKEN_URL + nextPageToken,
-              headers: {HttpHeaders.authorizationHeader: authToken});
+    _client == null
+        ? nextPageToken == null
```

Comment:
If url is conditional code is much cleaner.

Reply:
Done.